### ISSUE TITLE
[Feature]: support direnv in devbox wrapper shims

### ISSUE THREAD

**Issue by ahobson** [jetpack-io/devbox#1578]
**Is your feature request related to a problem you're trying to solve? Please describe.**

I'd like a way to modify the environment in `.envrc` files and have those settings be used when running commands installed via devbox.

```
$ devbox version
0.7.0
$ devbox init
$ devbox generate direnv
# ...
$ echo 'PATH_add ./scripts' >> .envrc
$ direnv allow
$ cat scripts/custom 
#!/bin/sh
echo running $0
$ custom
running /Users/ahobson/t/scripts/custom
$ devbox add overmind
# ...
$ overmind run custom
overmind: exec: "custom": executable file not found in $PATH
```

**Describe the solution you'd like**

Maybe the shims created by devbox that run `devbox shellenv` could detect if `DIRENV_FILE` is set and if so, run `direnv export` ? Or maybe some other custom setting in `devbox.json` ?

**Describe alternatives you've considered**

I'm honestly not sure of the best way of accomplishing this.

**Additional context**

**Comment by Lagoja** [comment]
Hey @ahobson, just to check if I understand the request -- you want the `.envrc` to be sourced _only_ on a script or in a shell, but not when `cd`'ing into the directory?

**Comment by ahobson** [comment]
@Lagoja The problem I'm seeing is that after `cd`ing into the directory, the `.envrc` modifies the `PATH`. But if I try to use a tool like `overmind`, installed via devbox, the `PATH` is overwritten by the `overmind` shims. The line in the shim that looks like `eval "$(DO_NOT_TRACK=1 devbox shellenv only-path-without-wrappers)"` overwrites any `PATH` modifications made by `.envrc`.

Does that make sense?

**Comment by Lagoja** [comment]
I think I understand, that should give me enough details to repro, so i'll test it out today

**Comment by Lagoja** [comment]
Hey @ahobson, I have a repro. I think this should be resolved by some work we have planned on shims for the upcoming cycle. I'll circle back once we have a testable version to see if it resolves this issue

**Comment by Lagoja** [comment]
Hey @ahobson, I think this issue is now fixed in our 0.8.0-dev pre release. Could you give it a test and see if it helps with the overmind issue?

To test the 0.8 prerelease, export the DEVBOX_USE_VERSION environment variable in your shell. We recommend adding this line to your shell's rc file

```bash
export DEVBOX_USE_VERSION=0.8.0-dev
```

To revert back to the previous release, unset this variable.

**Comment by ahobson** [comment]
@Lagoja Sorry for the delay, I was out of town. This works GREAT with devbox 0.8.1! Thank you very much!
